osu.Game/Online/Rooms/MultiplayerPlaylistItem.cs

```diff
@@ -82,5 +82,24 @@ public MultiplayerPlaylistItem(PlaylistItem item)
             StarRating = item.Beatmap.StarRating;
             Freestyle = item.Freestyle;
         }
+
+        public bool Equals(MultiplayerPlaylistItem? other)
+        {
+            if (other is null) return false;
+            if (ReferenceEquals(this, other)) return true;
+
+            return ID == other.ID
+                   && OwnerID == other.OwnerID
+                   && BeatmapID == other.BeatmapID
+                   && BeatmapChecksum.Equals(other.BeatmapChecksum, StringComparison.Ordinal)
```

Comment:
```suggestion
                   && BeatmapChecksum.Equals(other.BeatmapChecksum, StringComparison.OrdinalIgnoreCase)
```

for safety, probably